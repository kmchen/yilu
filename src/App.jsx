import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { Timeline, TimelineEvent } from 'react-event-timeline';

import Ground from './components/Ground';
import Takeoff from './components/TakeOff';
import Landing from './components/Landing';
import NavBar from './components/NavBar';
import convertDate from './util/util';

const useStyles = makeStyles(() => ({
  root: {
    backgroundColor: 'rgb(255, 234, 234)',
    height: '100vh',
    fontFamily: 'Roboto',
    fontSize: 15,
  },
  timelineEvent: {
    display: 'inline-block',
  },
}));

export function AppComponent({ itinerary }) {
  const classes = useStyles();
  const timelineEvents = itinerary.map(val => {
    let child;
    switch (val.type) {
      case 'GROUND':
        child = <Ground address={val.origin.value} />;
        break;
      case 'AIRPORT':
        child = (
          <Takeoff airportName={val.origin.value} busynessIndex={val.detail.busyness_index} />
        );
        break;

      case 'FLIGHT':
        child = (
          <Landing
            origin={val.origin.value}
            destination={val.destination.value}
            boardingTime={val.detail.boarding}
            flightNumber={val.detail.flight_number}
            isOnTime={val.detail.boarding - val.departure < 0}
            seat={val.detail.gate}
            gate={val.detail.seat}
          />
        );
        break;
      default:
        return null;
    }
    return (
      <TimelineEvent
        title=""
        createdAt={convertDate(val.departure)}
        icon={<i className="material-icons md-18">album</i>}
        className={classes.timelineEvent}
        key={val.type}
      >
        {child}
      </TimelineEvent>
    );
  });
  return (
    <div className={classes.root}>
      <NavBar />
      <Timeline>{timelineEvents}</Timeline>
    </div>
  );
}

/* eslint-disable react/forbid-prop-types */
AppComponent.propTypes = {
  itinerary: PropTypes.array.isRequired,
};

const mapStateToProps = state => {
  const { itinerary } = state;
  return { itinerary };
};

export const App = connect(
  mapStateToProps,
  null,
)(AppComponent);
