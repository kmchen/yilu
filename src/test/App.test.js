import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { AppComponent } from '../App';
import data from '../../data/data.js';

configure({ adapter: new Adapter() });

describe('App component', () => {
  let appComponent;

  beforeEach(() => {
    const props = {
      itinerary: data.segments,
    };
    appComponent = shallow(<AppComponent {...props} />);
  });

  it('should render App component', () => {
    expect(appComponent).toMatchSnapshot();
  });
});
