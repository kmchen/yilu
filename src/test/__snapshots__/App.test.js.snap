// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `
<div
  className="makeStyles-root-1"
>
  <NavBar />
  <Timeline
    lineStyle={Object {}}
    style={Object {}}
  >
    <TimelineEvent
      bubbleStyle={Object {}}
      cardHeaderStyle={Object {}}
      className="makeStyles-timelineEvent-2"
      contentStyle={Object {}}
      createdAt="7:20"
      icon={
        <i
          className="material-icons md-18"
        >
          album
        </i>
      }
      iconStyle={Object {}}
      key="GROUND"
      onClick={[Function]}
      onIconClick={[Function]}
      orientation="left"
      showContent={false}
      style={Object {}}
      subtitleStyle={Object {}}
      title=""
      titleStyle={Object {}}
    >
      <Ground
        address="Alte Schönhauser Straße 3, 10119 Berlin"
      />
    </TimelineEvent>
    <TimelineEvent
      bubbleStyle={Object {}}
      cardHeaderStyle={Object {}}
      className="makeStyles-timelineEvent-2"
      contentStyle={Object {}}
      createdAt="8:00"
      icon={
        <i
          className="material-icons md-18"
        >
          album
        </i>
      }
      iconStyle={Object {}}
      key="AIRPORT"
      onClick={[Function]}
      onIconClick={[Function]}
      orientation="left"
      showContent={false}
      style={Object {}}
      subtitleStyle={Object {}}
      title=""
      titleStyle={Object {}}
    >
      <Takeoff
        airportName="TXL"
        busynessIndex={10}
      />
    </TimelineEvent>
    <TimelineEvent
      bubbleStyle={Object {}}
      cardHeaderStyle={Object {}}
      className="makeStyles-timelineEvent-2"
      contentStyle={Object {}}
      createdAt="9:00"
      icon={
        <i
          className="material-icons md-18"
        >
          album
        </i>
      }
      iconStyle={Object {}}
      key="FLIGHT"
      onClick={[Function]}
      onIconClick={[Function]}
      orientation="left"
      showContent={false}
      style={Object {}}
      subtitleStyle={Object {}}
      title=""
      titleStyle={Object {}}
    >
      <Landing
        boardingTime={1523601000}
        destination="MUC"
        flightNumber="LH2031"
        gate="12C"
        isOnTime={true}
        origin="TXL"
        seat="C12"
      />
    </TimelineEvent>
  </Timeline>
</div>
`;
